### ISSUE TITLE
test one image

### ISSUE THREAD

**Issue by xiaohuaaibiancheng** [Ouxiang-Li/SAFE#5]
Why do I test fake images one by one, but mostly get a real label?

here is my test code:

`import torch
import json
from torchvision import transforms
from PIL import Image
import sys
import os

# Add model definition path (adjust according to actual path)
sys.path.append('./')
from models.resnet import ResNet, Bottleneck

def load_model(checkpoint_path):
    # Initialize model structure (must be consistent with training)
    model = ResNet(
        block=Bottleneck,
        layers=[3, 4, 6, 3],
        num_classes=2,  # Ensure consistency with training num_classes
        zero_init_residual=False
    )
    
    # Load checkpoint
    checkpoint = torch.load(checkpoint_path, map_location='cpu')
    
    # Handle possible distributed training save format
    state_dict = checkpoint.get('model', checkpoint)
    new_state_dict = {}
    for k, v in state_dict.items():
        name = k.replace('module.', '')  # Remove prefix from distributed training
        new_state_dict[name] = v
    
    model.load_state_dict(new_state_dict)
    model.eval()
    return model

def preprocess_image(image_path):
    # Preprocessing process (consistent with training)
    transform = transforms.Compose([
        transforms.Resize(256),
        transforms.CenterCrop(256),
        transforms.ToTensor(),
        transforms.Normalize(
            mean=[0.485, 0.456, 0.406],
            std=[0.229, 0.224, 0.225]
        ),
    ])
    
    image = Image.open(image_path).convert('RGB')
    return transform(image).unsqueeze(0)  # Add batch dimension

def predict(image_path, model):
    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
    model = model.to(device)
    
    input_tensor = preprocess_image(image_path).to(device)
    
    with torch.no_grad():
        output = model(input_tensor)
        probabilities = torch.softmax(output, dim=1)[0]
    
    return probabilities.cpu().numpy()

if __name__ == "__main__":
    # Parameter configuration
    checkpoint_path = './checkpoint/checkpoint-best.pth'  # Model checkpoint path
    image_path = r"D:\Desktop\OIP (4).jpg"           # Test image path

    # Load model
    model = load_model(checkpoint_path)
    
    # Execute prediction
    try:
        probs = predict(image_path, model)
        result = {
            "prediction": "fake" if probs[1] > 0.5 else "real",
            "confidence": {
                "real": float(probs[0]),
                "fake": float(probs[1])
            }
        }
    except Exception as e:
        result = {
            "error": str(e)
        }

    # Output JSON result
    print(json.dumps(result, indent=2))`

**Comment by SHYuanBest** [comment]
same issuse
